lm_eval/tasks/mslr2022/utils.py

```
@@ -0,0 +1,208 @@
+    config_name: 'cochrane' or 'ms2' (passed via dataset_kwargs)
+    """
+    config_name = kwargs.get("config_name", config_name)
+    cache_dir = os.path.expanduser("~/.cache/huggingface/datasets/mslr2022")
+    tar_path = os.path.join(cache_dir, "mslr_data.tar.gz")
+
+    mslr_data_dir = os.path.join(cache_dir, "mslr_data")
+    if not os.path.exists(mslr_data_dir):
+        eval_logger.info(f"MSLR2022 ({config_name}): Downloading and extracting dataset...")
+        os.makedirs(cache_dir, exist_ok=True)
+        if not os.path.exists(tar_path):
+            urlretrieve(_MSLR_URL, tar_path)
```

Ruff (0.15.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)